pkg/controller/cloudwatchlogs/loggroup/setup.go

```diff
@@ -0,0 +1,207 @@
+	if awsclients.Int64Value(cr.Spec.ForProvider.RetentionInDays) != awsclients.Int64Value(obj.LogGroups[0].RetentionInDays) {
+		return false, nil
+	}
+
+	tags, err := u.client.ListTagsLogGroup(&svcsdk.ListTagsLogGroupInput{
+		LogGroupName: awsclients.String(meta.GetExternalName(cr)),
+	})
+	if err != nil {
+		return false, errors.Wrap(err, errListTags)
+	}
+	add, remove := diffTags(cr.Spec.ForProvider.Tags, tags.Tags)
+
+	return len(add) == 0 && len(remove) == 0, nil
+}
+
+func (u *updater) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mg.(*svcapitypes.LogGroup)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	obj, err := u.client.DescribeLogGroupsWithContext(ctx, &svcsdk.DescribeLogGroupsInput{
+		LogGroupNamePrefix: awsclients.String(meta.GetExternalName(cr)),
+	})
+	if err != nil {
+		return managed.ExternalUpdate{}, awsclients.Wrap(err, errCreate)
+	}
+
+	tags, err := u.client.ListTagsLogGroup(&svcsdk.ListTagsLogGroupInput{
+		LogGroupName: awsclients.String(meta.GetExternalName(cr)),
+	})
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errListTags)
+	}
+	add, remove := diffTags(cr.Spec.ForProvider.Tags, tags.Tags)
+
+	if len(add) > 0 {
+		_, err := u.client.TagLogGroupWithContext(ctx, &svcsdk.TagLogGroupInput{
+			LogGroupName: awsclients.String(meta.GetExternalName(cr)),
+			Tags:         add,
+		})
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errTagResource)
+		}
+	}
+	if len(remove) > 0 {
+		_, err := u.client.UntagLogGroupWithContext(ctx, &svcsdk.UntagLogGroupInput{
+			LogGroupName: awsclients.String(meta.GetExternalName(cr)),
+			Tags:         remove,
+		})
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errUntagResource)
+		}
+	}
+
+	var zero int64 = 0
+	if cr.Spec.ForProvider.RetentionInDays == nil || awsclients.Int64Value(cr.Spec.ForProvider.RetentionInDays) == awsclients.Int64Value(&zero) {
+		if _, err := u.client.DeleteRetentionPolicy(&svcsdk.DeleteRetentionPolicyInput{
+			LogGroupName: awsclients.String(meta.GetExternalName(cr)),
+		}); err != nil {
+			return managed.ExternalUpdate{}, awsclients.Wrap(err, errUpdate)
+		}
+	}
+
+	if cr.Spec.ForProvider.RetentionInDays != nil &&
+		awsclients.Int64Value(cr.Spec.ForProvider.RetentionInDays) != awsclients.Int64Value(obj.LogGroups[0].RetentionInDays) {
+		if _, err := u.client.PutRetentionPolicy(&svcsdk.PutRetentionPolicyInput{
+			LogGroupName:    awsclients.String(meta.GetExternalName(cr)),
+			RetentionInDays: cr.Spec.ForProvider.RetentionInDays,
+		}); err != nil {
+			return managed.ExternalUpdate{}, awsclients.Wrap(err, errUpdate)
+		}
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+// diffTags returns tags that should be added or removed.
+func diffTags(spec map[string]*string, current map[string]*string) (addMap map[string]*string, remove []*string) {
```

Comment:
Looks like we have this variant another place too, consider if it would be possible to share this implementation, maybe in `pkg/clients/`: `pkg/controller/lambda/function/setup.go`

Reply:
moved diffTags `map[string]*string ` for ressources lambda, eks-addon and cloudwatch loggroup to aws clients and tested the creation, update , delete
thanks for pointing this out